src/display.rs

```diff
@@ -139,37 +180,105 @@ impl<'a, 'b> State<'a, 'b> {
             }
         }
     }
-    fn find_meta(&self, attrs: &[Attribute]) -> Result<Option<Meta>> {
-        let mut it = attrs
-            .iter()
-            .filter_map(|m| m.parse_meta().ok())
-            .filter(|m| {
-                if let Some(ident) = m.path().segments.first().map(|p| &p.ident) {
-                    ident == self.trait_attr
-                } else {
-                    false
+    fn find_meta(&self, attrs: &[syn::Attribute], meta_key: &str) -> Result<Option<syn::Meta>> {
+        let mut iterator = attrs.iter()
+            .filter_map(|attr| attr.parse_meta().ok())
+            .filter(|meta| {
+                let meta = match meta {
+                    syn::Meta::List(meta) => meta,
+                    _ => return false,
+                };
+
+                if !meta.path.is_ident(self.trait_attr) || meta.nested.is_empty() {
+                    return false;
                 }
+
+                let meta = match &meta.nested[0] {
+                    syn::NestedMeta::Meta(meta) => meta,
+                    _ => return false,
+                };
+
+                let meta = match meta {
+                    syn::Meta::NameValue(meta) => meta,
+                    _ => return false,
+                };
+
+                meta.path.is_ident(meta_key)
             });
 
-        let meta = it.next();
-        if it.next().is_some() {
-            Err(Error::new(meta.span(), "Too many formats given"))
-        } else {
+        let meta = iterator.next();
+        if iterator.next().is_none() {
             Ok(meta)
+        } else {
+            Err(Error::new(meta.span(), "Too many attributes specified"))
         }
     }
-    fn get_meta_fmt(&self, meta: &Meta, outer_enum: bool) -> Result<(TokenStream, bool)> {
+    fn parse_meta_bounds(&self, bounds: &syn::LitStr) -> Result<HashMap<syn::Type, HashSet<syn::TraitBound>>> {
+        let span = bounds.span();
+
+        let input = bounds.value();
+        let tokens = TokenStream::from_str(&input)?;
+        let parser = Punctuated::<syn::GenericParam, syn::Token![,]>::parse_terminated;
+        let bounds = parser.parse2(tokens).map_err(|error| Error::new(span, error.to_string()))?;
+
+        if bounds.is_empty() {
+            return Err(Error::new(span, "No bounds specified"));
+        }
+
+        bounds.into_iter().try_fold(HashMap::new(), |mut accumulator, generic_param| {
+            match generic_param {
+                syn::GenericParam::Type(type_param) => {
+                    if self.type_params.contains(&type_param.ident) {
+                        if !type_param.attrs.is_empty() {
+                            Err(Error::new(span, "Attributes aren't allowed"))
```

Comment:
Can you refactor this deeply nested code to use early returns instead? See this for a good example:
https://github.com/JelteF/derive_more/blob/master/src/utils.rs#L613-L633

Reply:
Done.